sdk/messaging/azservicebus/receiver.go

```diff
@@ -462,16 +474,24 @@ func fetchMessages(ctx context.Context, receiver internal.AMQPReceiver, maxMessa
 
 // flushPrefetchedMessages makes a best-effort attempt at flushing the internal channel for the link. This is
 // a purely in-memory operation so should work even if the link itself has failed.
-func flushPrefetchedMessages(ctx context.Context, receiver internal.AMQPReceiver, messages *[]*ReceivedMessage) {
+func flushPrefetchedMessages(receiver internal.AMQPReceiver, messages *[]*ReceivedMessage) {
+	count := 0
+	defer func() {
+		if count > 0 {
+			log.Writef(EventReceiver, "Flushed %d messages from receiver's internal cache", count)
+		}
+	}()
+
 	for {
-		am, err := receiver.Prefetched(ctx)
+		am, err := receiver.Prefetched(context.Background())
```

Comment:
I think we should change the `Prefetched()` API to not take a context.  It doesn't perform any IO and is always non-blocking.

Reply:
I'll file an issue in go-amqp for that one.

Reply:
https://github.com/Azure/go-amqp/issues/147